includes/wikia/models/DesignSystemGlobalNavigationModel.class.php

```diff
@@ -0,0 +1,277 @@
+				'header' => [
+					'type' => 'link-image',
+					'href' => $this->getHref( 'fandom-logo' ),
+					'image' => 'wds-company-logo-fandom',
+					'title' => [
+						'type' => 'text',
+						'value' => 'Fandom powered by Wikia'
+					]
+				]
+			],
+			'verticals' => [
+				'links' => [
+					[
+						'type' => 'link-branded',
+						'brand' => 'tv',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-brandlink-vertical-tv'
+						],
+						'href' => $this->getHref( 'tv' ),
+					],
+					[
+						'type' => 'link-branded',
+						'brand' => 'games',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-brandlink-vertical-games'
+						],
+						'href' => $this->getHref( 'games' ),
+					],
+					[
+						'type' => 'link-branded',
+						'brand' => 'movies',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-brandlink-vertical-movies'
+						],
+						'href' => $this->getHref( 'movies' ),
+					]
+				]
+			],
+			'wikis' => [
+				'header' => [
+					'type' => 'line-text',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-wikis',
+					],
+				],
+				'links' => [
+					[
+						'type' => 'link-text',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-brandlink-wikis-explore'
+						],
+						'href' => $this->getHref( 'explore-wikis' ),
+					],
+					[
+						'type' => 'link-text',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-wikis-communitycentral'
+						],
+						'href' => $this->getHref( 'community-central' ),
+					],
+					[
+						'type' => 'link-text',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-wikis-fandomuniversity'
+						],
+						'href' => $this->getHref( 'fandom-university' ),
+					]
+				]
+			],
+			'search' => [
+				'module' => [
+					'type' => 'search',
+					'results' => [
+						'url' => 'http://starwars.wikia.com/wiki/Special:Search?fulltext=Search',
+						'param' => 'query'
+					],
+					'suggestions' => [
+						'url' => 'http://starwars.wikia.com/index.php?action=ajax&rs=getLinkSuggest&format=json',
+						'param' => 'query'
+					],
+					'placeholder-inactive' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-search-placeholder-inactive'
+					],
+					'placeholder-active' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-search-placeholder-active'
+					]
+				]
+			],
+			'create_wiki' => [
+				'header' => [
+					'type' => 'link-text',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'wikia-create-wiki-link-start-wikia'
+					],
+					'href' => $this->getPageUrl( 'CreateNewWiki', NS_SPECIAL ),
+				]
+			]
+		];
+
+		if ($wgUser->isLoggedIn()) {
+			$data['user'] = $this->getLoggedInUserData( $wgUser->getId() );
```

Comment:
You can pass just `$wgUser`, so then you don't need to create `User` object inside function